### PR TITLE
Add system make installation check

### PR DESCRIPTION
**Pull request by ldotrg**
Give a user a prompt when system make not install yet.

### CODE REVIEW

File: scripts/init [hunk 1]
```diff
@@ -2,6 +2,20 @@ export DEVENVROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
 
 . ${DEVENVROOT}/scripts/env.d/vars
 
+#----------------------------------------------------------------------------------------------------
+#    make sure system make is installed and reachable 
+#----------------------------------------------------------------------------------------------------
+#  find make
+MAKE=$(which make 2>/dev/null)
+
+if (( $? == 1 )); then
```
Comment (yungyuc) [inline scripts/init:11]: I would like this to be fenced by an environment variable, like `$DEVENVVERBOSE` or `$DEVENVDEBUG`.  I think most of the time we would like `devenv` to be lean.

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: scripts/env.d/vars [hunk 2]
```diff
@@ -13,3 +13,17 @@ case "$(uname)" in
...
 # vim: set et nu nobomb fenc=utf8 ft=bash ff=unix sw=2 ts=2:
+
+#----------------------------------------------------------------------------------------------------
+#    make sure system make is installed and reachable 
+#----------------------------------------------------------------------------------------------------
+#  find make
+MAKE=$(which make 2>/dev/null)
+
+if (( $? == 1 )); then 
+  echo " "
```
Comment (tai271828) [inline scripts/env.d/vars:24]: `echo " "` echo an empty space is equivalent to `echo ""`. Do you prefer which one?

File: scripts/env.d/vars [hunk 3]
```diff
@@ -13,3 +13,17 @@ case "$(uname)" in
...
+  echo "ERROR: make needs to be installed to run devenv. "
+  echo " Install make using the command such as ' apt install build-essential' as root and try again. "
```
Comment (tai271828) [inline scripts/env.d/vars:26]: I am not sure the leading and trailing spaces are meant to be. Besides, `' apt install build-essential'` has a leading space as well. Are they typo or on purpose?

File: scripts/env.d/vars [hunk 4]
```diff
@@ -13,3 +13,17 @@ case "$(uname)" in
...
+  echo " "
```
Comment (tai271828) [inline scripts/env.d/vars:27]: ditto

File: scripts/env.d/vars [hunk 5]
```diff
@@ -13,3 +13,17 @@ case "$(uname)" in
...
+  exit
+fi
```
Comment (tai271828) [inline scripts/env.d/vars:29]: No newline at end of file

**Review by tai271828 — CHANGES_REQUESTED** [review]
Some coding style bike-shedding for your reference.

**Review by ldotrg — COMMENTED** [review]
> @ldotrg It seems to me that this PR still has unfinished work and can still be worked on. Why not keep working on it?
Instead of adding the warining message and fixing the "space". 
 I will add building GNU make from source. It's better than obsessed with some spaces and learned nothing in the end.
Because these typo problem could add the .clang_format to auto fix it.

**Pull request closed without merging**